Грехов Кирилл

Grehhov opened the pull request.

Grehhov committed:
Update phoneBook.js

mangin commented on phoneBook.js:
```diff
@@ -1,32 +1,74 @@
+			phone: phone,
+			email: email
+		});
+		console.log('Запись добавлена');
```
Логировать прекрасно, но лучше чтобы функция возвращала true/false в зависимости от успеха операции.
Так вы сможете обрабатывать ошибки пользовательского ввода и программа будет работать быстрее, так как вывод на консоль дорогая операция.

mangin commented on phoneBook.js:
```diff
@@ -1,32 +1,74 @@
 };
 
+function isPhone(phone) {
+	return /\+?\d? ?(\d{3}|\(\d{3}\))? ?\d{3}( |-)?\d+( |-)?\d{3}/.test(phone);
```
лучше просто вынести регулярки в константы и избавиться от этих функций

mangin commented on phoneBook.js:
```diff
@@ -1,32 +1,74 @@
 module.exports.find = function find(query) {
-
+	var contacts = findContact(query);
+	for(var item in contacts) {
```
бегать по массиву for in не безопасно, так как for in не гарантирует порядок, лучше бежать обычным циклом. Или воспользоваться for of, но for..of добавлен с es6 => нужно иметь довольно современный интерпретатор для вашего кода.

https://developer.mozilla.org/ru/docs/Web/JavaScript/Reference/Statements/for...of

mangin commented on phoneBook.js:
```diff
@@ -1,32 +1,74 @@
    Функция поиска записи в телефонную книгу.
    Поиск ведется по всем полям.
 */
 module.exports.find = function find(query) {
```
команда find должна возвращать массив записей, а не выводить их на экран.

mangin replied:
для вывода записи на экран лучше написать отдельную функцию

mangin commented on phoneBook.js:
```diff
@@ -1,32 +1,74 @@
+	var contacts = [];
+	for (var i = 0; i < phoneBook.length; i++) {
+		for (var item in phoneBook[i]) {
+			if (phoneBook[i][item].indexOf(query) != -1) {
```
!==

mangin commented on phoneBook.js:
```diff
@@ -1,32 +1,74 @@
+		for (var item in phoneBook[i]) {
+			var index = phoneBook[i][item].indexOf(query);
+			if (index !== -1) {
+				delete phoneBook[i];
```
в итоге вы получите разряженный массив => длинна исходного массива не будет уменьшаться.

mangin commented on phoneBook.js:
```diff
@@ -1,32 +1,74 @@
+			}
+		}
+	}
+	console.log('Удален(о) ' + count + ' контакт(ов)')
```
лучше возвращать список удаленных элементов, чтобы с ними можно было потом что-нибудь сделать.

mangin commented on phoneBook.js:
```diff
@@ -46,7 +88,13 @@ module.exports.importFromCsv = function importFromCsv(filename) {
 */
 module.exports.showTable = function showTable() {
-
+	var number = 0;
```
грустная ASCII функция.
В задании подразумевался вывод красивой таблички.

mangin commented on phoneBook.js:
```diff
@@ -46,7 +88,13 @@ module.exports.importFromCsv = function importFromCsv(filename) {
-
+	var number = 0;
+	for (var i = 0; i < phoneBook.length; i++) {
+		if (phoneBook[i] !== undefined) {
```
Это вы получили в результата работы с разряженным массивом

Grehhov committed:
Update phoneBook.js

@mangin

Grehhov committed:
Update phoneBook.js

@mangin, я не понял, как исправить последнее замечание, и for ... of не хочет работать с phoneBook[i]

mangin commented:
засчитано еще разок
